Deploy-CAM.ps1

```diff
@@ -3170,32 +3258,192 @@ if(-not $keyVaultProviderExists) {
+$CAMKeyVaults = Find-AzureRmResource -ResourceType "Microsoft.KeyVault/vaults" | Where-object {$_.Name -like "CAM-*"}
+
+$deploymentIndex = 0
+ForEach ($s in $CAMKeyVaults) {
+    if (-not (Get-Member -inputobject $s -name "Number")) {
+        Add-Member -InputObject $s -Name "Number" -Value "" -MemberType NoteProperty
+    }
+
+    $s.Number = ++$deploymentIndex
+}
+
+if((-not $ResourceGroupName) -and ($deploymentIndex -gt 0)) {
+    while ($true) {
+
+        Write-Host "`nCloud Access Manager deployments in this subscription"
+        Write-Host ($CAMKeyVaults | Select-Object -Property Number, ResourceGroupName | Format-Table | Out-String)
+    
+    
+        Write-Host ("To upgrade or modify a Cloud Access Manager Deployment, select the resource group of the`n" +
+           "Cloud Access Manager deployment root by number, or hit enter to create a new deployment.")
+        $rgIdentifier = (Read-Host "Resource group").Trim()
+        
+        # If empty string just exit loop
+        if(-not $rgIdentifier) {
+            break
+        }
+    
+        $rgIndex = 0
+        $rgIsInt = [int]::TryParse($rgIdentifier, [ref]$rgIndex) # rgIndex will be 0 on parse failure
+    
+        $rgArrayLength = $CAMKeyVaults.Length
+        if ( -not (( $rgIndex -ge 1) -and ( $rgIndex -le $rgArrayLength))) {
+            # Invalid range try again
+            Write-Host-Warning "Please enter a range between 1 and $rgArrayLength, or hit enter to create a new deployment"
+        }
+        else {
+            # Use the selected resource group for selection below
+            $ResourceGroupName = $CAMKeyVaults[$rgIndex - 1].ResourceGroupName
+            break
+        }
+    }
+}
+
+# Now we (might) have a $ResourceGroupName that (might) point to a resource group that has a CAM keyvault in it.
+# Let's double check what type.
+# At this point we have a subscription and a root resource group - check if there is already a deployment in it
+try{
+    $CAMRootKeyvault = $null
+    $CAMRootKeyvault = Get-AzureRmResource `
+        -ResourceGroupName $ResourceGroupName `
+        -ResourceType "Microsoft.KeyVault/vaults" `
+        -ErrorAction Stop `
+        | Where-object {$_.Name -like "CAM-*"}
+}
+catch {
+    # silently swallow - if we had an error we couldn't find the key vault, usually because the resource group didn't exist.
+}
+
+# If there is a root keyvault, verify there is only one and then query for connector deployment.
+if ($CAMRootKeyvault) {
+    if ($CAMRootKeyvault -is [Array]) {
+        Write-Host "More than one CAM Key Vault found in this resource group."
+        Write-Host "Please move or remove all but one."
+        return   # early return!
+    }
+
+    Write-Host "The resource group $ResourceGroupName has a CAM deployment."
+    Write-Host "Using key vault $($CAMRootKeyvault.Name)"
+
+    # Ensure this user account can get secrets from the key vault
+    $hasKVAccess = Set-KeyVaultAccess($CAMRootKeyvault)
```

Comment:
shouldn't this be 'Set-KeyVaultAccess -camRootKeyvalut $CAMRootKeyvault'?

Reply:
Yes. I think it worked because there was only one parameter but it is fixed.